packages/docusaurus-theme-classic/src/theme/DocSidebar/index.tsx

```diff
@@ -43,31 +47,67 @@ const isActiveSidebarItem = (item, activePath) => {
   return false;
 };
 
+type CollapseMap = Record<string, boolean | undefined>;
+
+const createInitialStates = (
+  items: readonly PropSidebarItem[],
+): CollapseMap => {
+  const map: Record<string, boolean | undefined> = {};
+  items.forEach((childItem: PropSidebarItem) => {
+    const itemType = childItem.type;
+    const itemLabel = childItem.label;
+    if (itemType === 'category') {
+      map[itemLabel] = (childItem as PropSidebarItemCategory).collapsed;
```

Comment:
using the category label as a key is not reliable, as this means 2 categories with the exact same label in the same sidebar would share the same collapsed state and be unexpectedly synced. It's not a very common sidebar config but we still want to avoid such bugs.

The sidebar collapsed states should rather be modeled in a recursive structure similar to the sidebar prop.

Reply:
also, you could probably leverage TS discriminated union types to avoid the `as PropSidebarItemCategory `
https://basarat.gitbook.io/typescript/type-system/discriminated-unions

Reply:
The active category should be initialized as `collapsed: false`